snuba_sdk/visitors.py

```diff
@@ -182,10 +184,15 @@ def _stringify_scalar(self, value: ScalarType) -> str:
         raise InvalidExpression(f"'{value}' is not a valid scalar")
 
     def _visit_column(self, column: Column) -> str:
-        alias_clause = ""
+        entity_alias_clause = ""
         if column.entity is not None and self.use_entity_aliases:
-            alias_clause = f"{column.entity.alias}."
-        return f"{alias_clause}{column.name}"
+            entity_alias_clause = f"{column.entity.alias}."
+
+        alias_clause = ""
+        if column.output_alias and self.use_output_aliases:
+            alias_clause = f" AS {column.output_alias}"
```

Comment:
Having to rely on state changes of the translator to decide if to apply the alias is also something you would not need to do if we had a specific type for AliasedExpressions